jsonnet/thanos-mixin/dashboards/sidecar.libsonnet

```diff
@@ -0,0 +1,163 @@
+local g = import '../lib/thanos-grafana-builder/builder.libsonnet';
+
+{
+  grafanaDashboards+:: {
+    'sidecar.json':
+      g.dashboard($._config.grafanaThanos.dashboardSidecarTitle)
+      .addTemplate('cluster', 'kube_pod_info', 'cluster', hide=if $._config.showMultiCluster then 0 else 2)
+      .addTemplate('namespace', 'kube_pod_info{%(clusterLabel)s="$cluster"}' % $._config, 'namespace')
+      .addRow(
+        g.row('Last Updated')
+        .addPanel(
+          g.panel('Successful Upload') +
+          g.tablePanel(
+            ['time() - max(thanos_objstore_bucket_last_successful_upload_time{namespace="$namespace",%(thanosSidecarSelector)s}) by (bucket)' % $._config],
+            {
+              Value: {
+                alias: 'Uploaded Ago',
+                unit: 's',
+                type: 'number',
+              },
+            },
+          )
+        )
+        .addPanel(
+          g.panel('Successful Hearthbeat') +
+          g.tablePanel(
+            ['time() - max(thanos_sidecar_last_heartbeat_success_time_seconds{namespace="$namespace",%(thanosSidecarSelector)s}) by (pod)' % $._config],
```

Comment:
This might be a good alerting rule for when the last heartbeat was more than 3min ago (duration made up). Not sure how helpful it is in this dashboard.